app/src/main/java/com/github/houseorganizer/houseorganizer/EventsAdapter.java

```diff
@@ -111,26 +112,43 @@ private void eventButtonListener(Event event, View v) {
                     db.collection("events")
                             .document(event.getId())
                             .delete();
+                    ArrayList<Event> newEvents = new ArrayList<>(calendar.getEvents());
+                    newEvents.remove(event);
+                    calendar.setEvents(newEvents);
+                    this.notifyItemRemoved(position);
                     dialog.dismiss();
                 })
                 .setNeutralButton(R.string.edit, (dialog, id) ->{
-                    LayoutInflater inflater = LayoutInflater.from(v.getContext());
-                    final View dialogView = inflater.inflate(R.layout.event_creation, null);
+                    final View dialogView = createEditDialog(v, event);
                     new AlertDialog.Builder(v.getContext())
                             .setTitle(R.string.event_editing_title)
                             .setView(dialogView)
-                            .setPositiveButton(R.string.confirm, (editForm, editFormId) -> editEventAndDismiss(event.getId(), editForm, dialogView, db))
+                            .setPositiveButton(R.string.confirm, (editForm, editFormId) -> editEventAndDismiss(event, editForm, dialogView, position))
                             .setNegativeButton(R.string.cancel, (editForm, editFormId) -> dialog.dismiss())
                             .show();
                 }).show();
     }
 
-    private void editEventAndDismiss(String eventId, DialogInterface editForm, View dialogView, FirebaseFirestore db) {
+    private View createEditDialog(View v, Event event) {
+        LayoutInflater inflater = LayoutInflater.from(v.getContext());
+        View retView = inflater.inflate(R.layout.event_creation, null);
```

Comment:
There is still a warning here about passing "null" as the view root

Reply:
Fixing this in next commit